modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridMapQueryExecutor.java

```diff
@@ -550,6 +617,15 @@ private void onQueryRequest0(
             else
                 releaseReservations(qctx);
 
+            // Log the error along with the SQL query and parameters.
+            if (qry != null) {
+                U.error(log, "Failed to execute local query: " + qry.query() + " with parameters: " + Arrays.toString(qry.parameters(params)) + ". Node ID: " + node.id(), e);
+            } else {
+                U.error(log, "Failed to execute local query and query details are not available", e);
+                throw new IgniteCheckedException("Error executing query due to internal error.", e);
```

Comment:
This might be problematic because this class retries the request and/or remaps it to different nodes whenever possible.
Throwing the exception here cancels all possible further processing.

Reply:
Refactored the code. The logging is performed without prematurely halting the processing.